satpy/readers/msi_safe.py

```diff
@@ -59,31 +58,23 @@ def get_dataset(self, key, info):
             return
 
         logger.debug('Reading %s.', key['name'])
-        # FIXME: get this from MTD_MSIL1C.xml
-        quantification_value = 10000.
-        jp2 = glymur.Jp2k(self.filename)
-        bitdepth = 0
-        for seg in jp2.codestream.segment:
-            try:
-                bitdepth = max(bitdepth, seg.bitdepth[0])
-            except AttributeError:
-                pass
-
-        jp2.dtype = (np.uint8 if bitdepth <= 8 else np.uint16)
-
-        # Initialize the jp2 reader / doesn't work in a multi-threaded context.
-        # jp2[0, 0]
-        # data = da.from_array(jp2, chunks=CHUNK_SIZE) / quantification_value * 100
-
-        data = da.from_delayed(delayed(jp2.read)(), jp2.shape, jp2.dtype)
-        data = data.rechunk(CHUNK_SIZE) / quantification_value * 100
-
-        proj = DataArray(data, dims=['y', 'x'])
+        proj = self._read_from_file()
         proj.attrs = info.copy()
         proj.attrs['units'] = '%'
         proj.attrs['platform_name'] = self.platform_name
         return proj
 
+    def _read_from_file(self):
+        proj = rioxarray.open_rasterio(self.filename, chunks=CHUNK_SIZE)
+        return self._calibrate(proj.squeeze("band"))
+
+    @staticmethod
+    def _calibrate(proj):
+        proj = proj.where(proj > 0)
+        # FIXME: get this from MTD_MSIL1C.xml
```

Comment:
Since you're always telling me...this should probably be made into an issue right?

Reply:
yep thanks, creating issue now